.github/CODEOWNERS

```diff
@@ -82,7 +82,7 @@
 # layer, despite being located in the `sql` crate.
 /src/sql/src/plan/lowering.rs       @MaterializeInc/compute
 /src/sql-lexer                      @MaterializeInc/adapter
-/src/sql-parser                     @MaterializeInc/adapter
+/src/sql-parser                     @MaterializeInc/adapter @MaterializeInc/devex
```

Comment:
@morsapaes could you add the same note from the PR description to a comment in this file? Namely that it's the adapter team that actually owns the code in the SQL parser, and DevEx is here primarily for notification purposes.